lib/elixir/test/elixir/kernel/def_test.exs

```diff
@@ -0,0 +1,25 @@
+Code.require_file("../test_helper.exs", __DIR__)
+
+defmodule Kernel.DefTest do
+  use ExUnit.Case, async: true
+
+  def fun_with_block_defaults(
+        x,
+        y \\ (
+          default = "y"
+          default
+        ),
+        z \\ (
+          default = "z"
+          default
+        )
+      ) do
+    {x, y, z}
+  end
+
+  test "function with block defaults" do
+    assert {1, "y", "z"} = fun_with_block_defaults(1)
+    assert {1, 2, "z"} = fun_with_block_defaults(1, 2)
+    assert {1, 2, 3} = fun_with_block_defaults(1, 2, 3)
+  end
```

Comment:
Let's also add a test that accessing a variable from a default block raises. Something like this:

```
def test(_ \\ (default = 1)), do: default
```

You can use something like:

assert_raise ..., fn ->
  defmodule VarDefaultScope do
    def test(_ \\ (default = 1)), do: default
  end
end
```

Reply:
Added in [3cdb863](https://github.com/elixir-lang/elixir/pull/11758/commits/3cdb863f35cdab1ce368a5bdfb1f53ac057eee55)